dex/networks/eth/contracts/ETHSwapV0.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+pragma solidity >=0.7.0 <0.9.0;
+contract ETHSwap {
+    enum State { Empty, Filled, Redeemed, Refunded }
+
+    struct Swap {
+        uint initBlockNumber;
+        uint refundBlockTimestamp;
+        bytes32 secretHash;
+        bytes32 secret;
+        address initiator;
+        address participant;
+        uint256 value;
+        State state;
+    }
+
+    mapping(bytes32 => Swap) public swaps;
+
+    constructor() {}
+
+    modifier isRefundable(bytes32 secretHash, address refunder) {
+        require(swaps[secretHash].state == State.Filled);
+        require(swaps[secretHash].initiator == refunder);
+        uint refundBlockTimestamp = swaps[secretHash].refundBlockTimestamp;
+        require(block.timestamp >= refundBlockTimestamp);
+        _;
+    }
+
+    modifier isRedeemable(bytes32 secretHash, bytes32 secret, address redeemer) {
+        require(swaps[secretHash].state == State.Filled);
+        require(swaps[secretHash].participant == redeemer);
+        require(sha256(abi.encodePacked(secret)) == secretHash);
+        _;
+    }
+
+    modifier isNotInitiated(bytes32 secretHash) {
+        require(swaps[secretHash].state == State.Empty);
```

Comment:
curious why checking like this when I feel `swaps[secretHash]` should just be verified to be null